### ISSUE TITLE
npm ls: throws error for strongloop package

### ISSUE THREAD

**Issue by cebor** [strongloop/strongloop#268]
Hi,

getting following errors with the strongloop pack and npm@3:

```
bash
$ npm ls -g --depth=0
/usr/local/lib
├── npm@3.3.10
└── strongloop@5.0.1

npm ERR! extraneous: jsonparse@1.0.0 /usr/local/lib/node_modules/strongloop/node_modules/jsonparse
npm ERR! missing: jsonstream@1.0.3, required by jayson@1.2.1
```

The second error maybe is related to https://github.com/tedeh/jayson/pull/58.

**Comment by cebor** [comment]
Ok think the problem is that, osx is case-insensitive and the new "flat" design of npm@3.

https://www.npmjs.com/package/jsonstream
https://www.npmjs.com/package/JSONStream

Same packages with different case-sensitivity, and also a different version. 1.0.3 and 1.0.6 ^^

**Comment by rmg** [comment]
There is a PR open to update some of the optional dependencies: #266 

You could give it a try and see how it fairs: `npm install -g strongloop/strongloop#updated-hoisting`

**Comment by cebor** [comment]
same error:

```bash
$ npm ls -g --depth=0
/usr/local/lib
├── npm@3.3.11
└── strongloop@5.0.1 (git://github.com/strongloop/strongloop.git#896ec3731ab5593e3d455495c5c84886bcc89bab)

npm ERR! extraneous: jsonparse@1.0.0 /usr/local/lib/node_modules/strongloop/node_modules/jsonparse
npm ERR! missing: jsonstream@1.0.3, required by jayson@1.2.1
```

**Comment by surfjedi** [comment]
+1

**Comment by rmg** [comment]
Looks like tedeh/jayson#58 has been landed and released as jayson@1.2.2, so that takes care of our final jsonstream/JSONStream warning :-)

**Comment by rmg** [comment]
@cebor the jsonstream/JSONStream has been fixed, and I can't reproduce the extraneous jsonparse issue, so I'm going to close this now. Thanks for reporting it :-)
